src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -187,33 +190,35 @@ public void didChange(DidChangeTextDocumentParams params) {
         File tempFile = null;
 
         try {
-            tempFile = File.createTempFile("smithy", SmithyProject.SMITHY_EXTENSION);
+            if (params.getContentChanges().size() > 0) {
```

Comment:
Without this, `tempFile` would be set **before** `.get(0)` blows up, and `Option.ofNullable` will not be reliable (it will have a filename, but the file will be empty)